core/src/main/java/com/microsoft/applicationinsights/internal/persistence/LocalFileLoader.java

```diff
@@ -0,0 +1,110 @@
+package com.microsoft.applicationinsights.internal.persistence;
+
+import com.microsoft.applicationinsights.TelemetryChannel;
+import com.microsoft.applicationinsights.internal.config.connection.EndpointProvider;
+import com.microsoft.applicationinsights.internal.util.ThreadPoolUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.applicationinsights.internal.persistence.PersistenceHelper.TEMPORARY_FILE_EXTENSION;
+
+/**
+ * This class manages loading a list of {@link ByteBuffer} from the disk.
+ */
+public class LocalFileLoader {
+
+    private static final Logger logger = LoggerFactory.getLogger(LocalFileLoader.class);
+    private static final long INTERVAL = TimeUnit.SECONDS.toSeconds(30); // send persisted telemetries from local disk every 30 seconds.
```

Comment:
I think it was mentioned that other SDKs don't do this, and instead check disk after each successful send?

Reply:
yeah, but raj proposed this is better.  this is how he implemented for otel-dotnet exporter.  we can talk about it. when there is no offline storage, it does nothing.